File: backend/src/modules/classes/domain/enrollment.ts
export const ENROLLMENT_STATUSES = ['ACTIVE', 'WITHDRAWN'] as const;
export type EnrollmentStatus = (typeof ENROLLMENT_STATUSES)[number];

export type Enrollment = {
  id: string;
  teacherId: string;
  classId: string;
  studentId: string;
  status: EnrollmentStatus;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date | null;
};

/** Matrícula enriquecida com dados do aluno, usada em listagens por turma. */
export type EnrollmentWithStudent = Enrollment & {
  student: {
    id: string;
    name: string;
    registryCode: string | null;
    email: string | null;
  };
};

File: frontend/lib/domain/entities/enrollment.dart
import 'student.dart';

class Enrollment {
  const Enrollment({
    required this.id,
    required this.classId,
    required this.studentId,
    required this.status,
    this.student,
  });

  final String id;
  final String classId;
  final String studentId;
  final String status;
  final Student? student;

  bool get isActive => status == 'ACTIVE';
}
